### ISSUE TITLE
`Style/RedundantParentheses` autocorrects code into invalid Ruby

### ISSUE THREAD

**Issue by a-lavis** [rubocop/rubocop#12298]
`Style/RedundantParentheses` can sometimes autocorrect code into invalid Ruby.
This can happen when passing in an argument to a function that consists of calling a method on a hash.

--------

## Expected behavior

RuboCop should not autocorrect in this case.  I also think the `Style/RedundantParentheses` should probably ignore this scenario as well.

## Actual behavior

It autocorrects this case, and thus produced invalid Ruby (which RuboCop cannot parse).

When running `rubocop -A --debug test.rb` (see the below example from the "Steps to reproduce the problem" section), I see:
```
For /Users/aidanlavis: Default configuration from /Users/aidanlavis/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.57.1/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /Users/aidanlavis/test.rb
Loading cache from /Users/aidanlavis/.cache/rubocop_cache/33ac87dfe8652207bb74821fb2c25f70e384417f/6d7a3b621ca1730e04accd938619e4bdab66cfb1/75f1154f8ccaf01f5993674deb19cb9ba5e3d00c
F

Offenses:

test.rb:3:6: C: [Corrected] Style/RedundantParentheses: Don't use parentheses around a literal.
puts ({ a: 1, b: 2 }).merge({ c: 3 })
     ^^^^^^^^^^^^^^^^
test.rb:3:9: F: Lint/Syntax: unexpected token tCOLON
(Using Ruby 2.7 parser; configure using TargetRubyVersion parameter, under AllCops)
puts { a: 1, b: 2 }.merge({ c: 3 })
        ^

1 file inspected, 2 offenses detected, 1 offense corrected
Finished in 0.08834399999977904 seconds
```

## Steps to reproduce the problem

Create a file `test.rb`
```ruby
# frozen_string_literal: true

puts ({ a: 1, b: 2 }).merge({ c: 3 })
```
You can run this file `ruby test.rb` and nothing breaks.
If you run `rubocop -A test.rb`, it autocorrects the file to:
```ruby
# frozen_string_literal: true

puts { a: 1, b: 2 }.merge({ c: 3 })
```
This file cannot be parsed - if you try to run it by doing `ruby test.rb` or if you try to run rubocop on it (`rubocop test.rb`) it will fail.

## RuboCop version

```
> rubocop -V
1.57.1 (using Parser 3.2.2.4, rubocop-ast 1.29.0, running on ruby 3.2.2) [arm64-darwin22]
```
